core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java

```diff
@@ -561,9 +562,12 @@ private void openBlock(BlockReadRequestContext context)
       }
       BlockReader reader = mWorker.createBlockReader(request);
       context.setBlockReader(reader);
-      if (reader instanceof UnderFileSystemBlockReader) {
```

Comment:
Not quite sure is it good to maintain the original one as well, this implementation kind of depending on the createBlockReader impl, whether or not it returns a DelegatingBlockReader or an UnderFileSystemBlockReader directly.
@apc999 what's your opinion?

Reply:
I fixed it to check for both